### PR TITLE
Add Pairtools parse

### PR DESCRIPTION
**Pull request by diegomics**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/pairtools_parse/macros.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<macros>
+    <token name="@TOOL_VERSION@">1.0.3</token>
+    <token name="@SUFFIX_VERSION@">0</token>
```
[the remark this answers is not in the document]
Comment (SaimMomin12) [reply tools/pairtools_parse/macros.xml:3]:
```suggestion
    <token name="@VERSION_SUFFIX@">0</token>
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<tool id="Pairtools_parse" name="Pairtools parse" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
```
[the remark this answers is not in the document]
Comment (SaimMomin12) [reply tools/pairtools_parse/pairtools_parse.xml:1]:
```suggestion
<tool id="pairtools_parse" name="Pairtools parse" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <description>Find ligation pairs in .sam data, make .pairs.</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="xrefs"/>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        Pairtools parse
```
[the remark this answers is not in the document]
Comment (SaimMomin12) [reply tools/pairtools_parse/pairtools_parse.xml:8]:
```suggestion
        pairtools parse
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            '$SAM_PATH'
+            -c '$chroms_path'
+            --min-mapq '$minimal_MAPQ'
+            --walks-policy '$walks_policy'
+            --max-inter-align-gap '$max_int_algn_gap'
+            --nproc-in ${GALAXY_SLOTS:-4}
+            --nproc-out ${GALAXY_SLOTS:-4}
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:15]:
```suggestion
            --nproc-in \${GALAXY_SLOTS:-4}
            --nproc-out \${GALAXY_SLOTS:-4}
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    ]]></command>
+    <inputs>
+        <param name="SAM_PATH" type="data" format="sam,bam,SAM,BAM" label="Input SAM/BAM file" help="Input .sam or .bam file with paired-end sequence alignments of Hi-C molecules."/>
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:18]:
```suggestion
        <param name="SAM_PATH" type="data" format="sam,bam" label="Input SAM/BAM file" help="Input .sam or .bam file with paired-end sequence alignments of Hi-C molecules."/>    
```
Are those BAM files name or coordinate sorted?

File: tools/pairtools_parse/pairtools_parse.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            -c '$chroms_path'
+            --min-mapq '$minimal_MAPQ'
+            --walks-policy '$walks_policy'
+            --max-inter-align-gap '$max_int_algn_gap'
+            --nproc-in ${GALAXY_SLOTS:-4}
+            --nproc-out ${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="SAM_PATH" type="data" format="sam,bam,SAM,BAM" label="Input SAM/BAM file" help="Input .sam or .bam file with paired-end sequence alignments of Hi-C molecules."/>    
+        <param name="chroms_path" type="data" format="txt,genome,sizes" label="Input a chromosome order file" help="Chromosome order used to flip interchromosomal mates. Any scaffolds not listed will be ordered lexicographically."/>
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:19]: This needs to be proper Galaxy datatypes.
```suggestion
        <param name="chroms_path" type="data" format="txt,tabular" label="Input a chromosome order file" help="Chromosome order used to flip interchromosomal mates. Any scaffolds not listed will be ordered lexicographically."/>
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        <param name="minimal_MAPQ" type="integer" value="40" min="0" label="Minimal MAPQ" help="Minimal MAPQ score to consider a read as uniquely mapped."/>
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:20]:
```suggestion
        <param name="minimal_mapq" type="integer" value="40" min="0" label="Minimal MAPQ" help="Minimal MAPQ score to consider a read as uniquely mapped."/>    
```

File: tools/pairtools_parse/pairtools_parse.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        <param name="SAM_PATH" type="data" format="sam,bam,SAM,BAM" label="Input SAM/BAM file" help="Input .sam or .bam file with paired-end sequence alignments of Hi-C molecules."/>    
+        <param name="chroms_path" type="data" format="txt,genome,sizes" label="Input a chromosome order file" help="Chromosome order used to flip interchromosomal mates. Any scaffolds not listed will be ordered lexicographically."/>
+        <param name="minimal_MAPQ" type="integer" value="40" min="0" label="Minimal MAPQ" help="Minimal MAPQ score to consider a read as uniquely mapped."/>    
+        <param name="walks_policy" type="select" label="Walks Policy" help="The policy for reporting unrescuable walks.">
+            <expand macro="walks_policy_options"/>
+        </param>   
+        <param name="max_int_algn_gap" type="integer" value="30" min="0" label="Max alignment gap" help="read segments that are not covered by any alignment and longer than the specified value are treated as null alignments."/>
+    </inputs>
+    <outputs>
+        <data name="output_parsed_pairs" format="pairs" label="${tool.name} on ${on_string}: .pairs"/>
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:27]: Is the format tabular?

File: tools/pairtools_parse/pairtools_parse.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    </tests>
+    <help><![CDATA[
+        **Pairtools parse**
+
+        Detects ligation events in the aligned sequences of DNA molecules formed in Hi-C experiments and reports them in the .pairs/.pairsam format.
+        
+    ]]></help>
+    <stdio>
+        <exit_code range="1:" level="fatal"/>
+    </stdio>
```
Comment (SaimMomin12) [inline tools/pairtools_parse/pairtools_parse.xml:48]:
```suggestion
```
Since you are using `detect_errors="exit_code"` you dont it.

**Review by SaimMomin12 — COMMENTED** [review]
Thanks @diegomics a few comments inline.

I think it would be good to convert this into s suite of tools. For this you can rename the folder to `pairtools` and change your shed.yml to a structure that looks like https://github.com/galaxyproject/tools-iuc/blob/main/tools/anndata/.shed.yml

File: tools/pairtools/parse.xml [hunk 10]
```diff
@@ -0,0 +1,127 @@
+<tool id="pairtools_parse" name="Pairtools parse" version="@TOOL_VERSION@+galaxy@SUFFIX_VERSION@" profile="22.05" license="MIT">
+    <description>Find ligation pairs in .sam data, make .pairs.</description>
```
Comment (bgruening) [inline tools/pairtools/parse.xml:2]:
```suggestion
    <description>Find ligation pairs in alignments and create pairs</description>
```

File: tools/pairtools/parse.xml [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+            #if $output_stats:
+                '$parsed_pairs_stats'
+            #end if
+            --walks-policy '$walks_policy'
+            --max-inter-align-gap '$max_inter_algn_gap'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="sam_path" type="data" format="sam,bam" label="Input SAM/BAM file" help="Input .sam or .bam file with paired-end sequence alignments of Hi-C molecules."/>
```
Comment (bgruening) [inline tools/pairtools/parse.xml:28]:
```suggestion
        <param name="sam_path" type="data" format="sam,bam" label="Input SAM/BAM file" help="Input SAM or BAM file with paired-end sequence alignments of Hi-C molecules."/>    
```

**Comment by bgruening** [comment]
The tests are only failing because the SAM header is changing all the time. This is expected, please use lines_diff in your tests to accommodate for this.

**Comment by bgruening** [comment]
Look very good to me. I added a commit to use `argument=""` wherever we can and also sorted the attributes (this can be cone easily with VSC) for better readability.

However, its untested, I hope it turns still green :)

**Review by bgruening — APPROVED** [review]

**Pull request merged**